chain/deals/provider_utils.go

```diff
@@ -80,12 +78,14 @@ func (p *Provider) sendSignedResponse(resp *Response) error {
 		return xerrors.Errorf("serializing response: %w", err)
 	}
 
-	worker, err := p.getWorker(p.actor)
+	ctx := context.TODO()
```

Comment:
rather than context.TODO() here I would put this as a parameter to sendSignedResponse since a context is available when we call it. 

Ideally let's not drop contexts cause someone may want to trace execution one day.